src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -993,7 +994,13 @@ void NetworkOPsImp::apply (std::unique_lock<std::mutex>& batchLock)
                 for (auto const& tx : m_ledgerMaster.pruneHeldTransactions(
                     txCur->getAccountID(sfAccount), txCur->getSequence() + 1))
                 {
-                    submitTransaction(tx);
+                    std::string reason;
+                    auto const trans = sterilize(*tx);
+                    auto t = std::make_shared<Transaction>(
+                        trans, reason, app_);
+                    submit_held.emplace_back(
+                        t, false, false, FailHard::no);
+                    t->setApplying();
```

Comment:
It seems like we shouldn't need all this. We should just be able to put the transaction back in the queue without this. For example, we shouldn't have held the transaction without already sterilizing it because we hold a transaction when it can't be applied immediately, and we shouldn't have even checked that before sterilizing. (We can come back and try to optimize this later though. It's already better than it was.)